File: ops/backend/app/schemas/risk.py
"""Risk schemas."""

from typing import Literal

from pydantic import BaseModel

from app.schemas.common import CategoryValue


class RiskLimit(BaseModel):
    label: str
    used: float | None = None
    limit: float | None = None
    unit: Literal["currency", "percent"]


class RiskData(BaseModel):
    """Aggregated risk posture across the book."""

    dailyLoss: RiskLimit
    weeklyLoss: RiskLimit
    monthlyLoss: RiskLimit
    currentExposure: float | None = None
    maxExposure: float | None = None
    currentMargin: float | None = None
    marginLevelPct: float | None = None
    currentDrawdownPct: float | None = None
    maxDrawdownPct: float | None = None
    valueAtRisk: float | None = None
    exposureBySymbol: list[CategoryValue]
    exposureByStrategy: list[CategoryValue]
    exposureByBroker: list[CategoryValue]

File: ops/frontend/src/types/risk.ts
import type { CategoryValue } from './common'

/** A consumed-vs-limit gauge (loss limits, exposure caps). */
export interface RiskLimit {
  label: string
  /** Amount consumed so far (absolute, same unit). */
  used: number
  /** Hard limit. */
  limit: number
  unit: 'currency' | 'percent'
}

/** Aggregated risk posture across the book. */
export interface RiskData {
  dailyLoss: RiskLimit
  weeklyLoss: RiskLimit
  monthlyLoss: RiskLimit
  /** Notional exposure currently deployed. */
  currentExposure: number
  /** Configured maximum notional exposure. */
  maxExposure: number
  /** Margin currently posted. */
  currentMargin: number
  marginLevelPct: number
  currentDrawdownPct: number
  maxDrawdownPct: number
  /** 1-day 95% value at risk. */
  valueAtRisk: number
  exposureBySymbol: CategoryValue[]
  exposureByStrategy: CategoryValue[]
  exposureByBroker: CategoryValue[]
}
